etl/steps/data/garden/wb/2023-08-14/shared.py

```diff
@@ -0,0 +1,764 @@
+            tb_garden[col_name].metadata = var_metadata_between_absolute_povlines(
+                var, povline_list[4], povline_list[6], origins, ppp_version, welfare_type
+            )
+
+        for rel in rel_dict:
+            # For variables that use relative poverty lines
+            col_name = f"{var}_{rel}_median"
+
+            if col_name in cols:
+                # Get the origins of the variable
+                origins = tb_garden[col_name].metadata.origins
+
+                # Create metadata for these variables
+                tb_garden[col_name].metadata = var_metadata_relative_povlines(var, rel, origins, welfare_type)
+
+                # Replace placeholders
+                tb_garden[col_name].metadata.description_short = (
+                    tb_garden[col_name]
+                    .metadata.description_short.replace("{povline}", rel_dict[rel])
+                    .replace("{inc_cons_dict[wel]['name']}", inc_cons_dict[welfare_type]["name"])
+                )
+
+        for pct in pct_dict:
+            # For variables that use percentiles (deciles)
+            col_name = f"decile{pct}_{var}"
+
+            if col_name in cols:
+                # Get the origins of the variable
+                origins = tb_garden[col_name].metadata.origins
+
+                # Create metadata for these variables
+                tb_garden[col_name].metadata = var_metadata_percentiles(var, pct, origins, ppp_version, welfare_type)
+
+                # Replace placeholders
+                tb_garden[col_name].metadata.description_short = (
+                    tb_garden[col_name]
+                    .metadata.description_short.replace("{str(pct)}", str(pct))
+                    .replace(
+                        "{inc_cons_dict[wel]['name_distribution']}",
+                        inc_cons_dict[welfare_type]["name_distribution"],
+                    )
+                    .replace("{inc_cons_dict[wel]['verb']}", inc_cons_dict[welfare_type]["verb"])
+                    .replace("{pct_dict[pct]['decile10']}", pct_dict[pct]["decile10"].lower())
+                )
+
+                tb_garden[col_name].metadata.unit = tb_garden[col_name].metadata.unit.replace("{ppp}", str(ppp_version))
+
+    return tb_garden
+
+
+# Metadata functions to show a clearer main code
+def var_metadata_inequality_mean_median(var, origins, welfare_type) -> VariableMeta:
+    """
+    Create metadata for defined uniquely by their name
+    """
+    # For monetary variables I include PPP description
+    if var in ["mean", "median"]:
+        meta = VariableMeta(
+            title=f"{var_dict[var]['title']} {inc_cons_dict[welfare_type]['name']}",
+            description_short=var_dict[var]["description"],
+            description_key=[
+                ppp_description,
+                inc_cons_dict[welfare_type]["description"],
+                non_market_income_description,
+            ],
+            description_processing=f"""
```

Comment:
@Marigold Hi! Do you know the reason why `description_processing` is not being saved in here or in any od the other functions below? All the others are saved correcly. I even changed it for a `"Hello"` here and still nothing.

Reply:
It works for me. Maybe you could try changing this line to "Hello" and then description_processing under `else:` branch below to "World" and see what you get.

Reply:
Right. I saw it dissapears because I do further processing to the table, extending it (I didn't know the ETL does that). But there's still a weird behavior: it's saving the wrong `description_processing` for the tables `income_2011` or `income_2017`. At least in my machine it's showing
```
To construct a global dataset, the World Bank combines estimates based on income data and estimates based on consumption data. Here we only include the estimates based on CONSUMPTION data.

You can find the data with all available income and consumption data points in our complete dataset of the World Bank PIP data.
```

When instead of CONSUMPTION it should show INCOME. Do you have it as well?

Reply:
I think it must be something with copying tables. Somewhere you get a subset (slice) of the table and don't call `.copy()` and then both tables have the same metadata object which overwrites itself.

[This place is suspicious](https://github.com/owid/etl/pull/1505/files#diff-a7cc65423e4ef280f1bfe78bffd3143180ca2061a17375fa2ab3d5e44150cddaR686). You should change it to this
```
tb_inc = tb[tb["welfare_type"] == "income"].reset_index(drop=True).copy()
tb_cons = tb[tb["welfare_type"] == "consumption"].reset_index(drop=True).copy()
```
it's possible there are other places.

Reply:
That was the one! I corrected it in some other places. So the recommended action to take is always to copy the table in a function or only when I am creating a different table?

Reply:
Great, I'm glad it worked :). You need it only when you create a different table.